src/deepsparse/benchmark/benchmark_pipeline.py

```diff
@@ -0,0 +1,525 @@
+    )
+    parser.add_argument(
+        "-ncores",
+        "--num_cores",
+        type=int,
+        default=cpu_architecture().num_available_physical_cores,
+        help=(
+            "The number of physical cores to run the analysis on, "
+            "defaults to all physical cores available on the system"
+        ),
+    )
+    parser.add_argument(
+        "-s",
+        "--scenario",
+        type=str,
+        default="sync",
+        choices=["async", "sync", "elastic"],
+        help=(
+            "Choose between using the async, sync and elastic scenarios. Sync and "
+            "async are similar to the single-stream/multi-stream scenarios. Elastic "
+            "is a newer scenario that behaves similarly to the async scenario "
+            "but uses a different scheduling backend. Default value is sync."
+        ),
+    )
+    parser.add_argument(
+        "-t",
+        "--time",
+        type=int,
+        default=10,
+        help="The number of seconds the benchmark will run. Default is 10 seconds.",
+    )
+    parser.add_argument(
+        "-w",
+        "--warmup_time",
+        type=int,
+        default=2,
+        help=(
+            "The number of seconds the benchmark will warmup before running."
+            "Default is 2 seconds."
+        ),
+    )
+    parser.add_argument(
+        "-nstreams",
+        "--num_streams",
+        type=int,
+        default=None,
+        help=(
+            "The number of streams that will submit inferences in parallel using "
+            "async scenario. Default is automatically determined for given hardware "
+            "and may be sub-optimal."
+        ),
+    )
+    parser.add_argument(
+        "-pin",
+        "--thread_pinning",
+        type=str,
+        default="core",
+        choices=["none", "core", "numa"],
+        help=(
+            "Enable binding threads to cores ('core' the default), "
+            "threads to cores on sockets ('numa'), or disable ('none')"
+        ),
+    )
+    parser.add_argument(
+        "-e",
+        "--engine",
+        type=str,
+        default=DEEPSPARSE_ENGINE,
+        help=(
+            "Inference engine backend to run eval on. Choices are 'deepsparse', "
+            "'onnxruntime'. Default is 'deepsparse'. Can also specify a user "
+            "defined engine class by giving the script and class name in the "
+            "following format <path to python script>:<Engine Class name>. This "
+            "engine class will be dynamically imported during runtime"
+        ),
+    )
+    parser.add_argument(
+        "-q",
+        "--quiet",
+        help="Lower logging verbosity",
+        action="store_true",
+        default=False,
+    )
+    parser.add_argument(
+        "-x",
+        "--export_path",
+        help="Store results into a JSON file",
+        type=str,
+        default=None,
+    )
+
+    return parser.parse_args()
+
+
+class PipelineExecutorThread(threading.Thread):
```

Comment:
Could you add docstring to this class and its functions?

Reply:
Can you also add a quick diagram/docstring showing the lifecycle of this pipeline?